# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xdialog/combobox-default-item.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Allowing to set a default entry for the combo box is often a must.

  - Rene Rebe <rene@exactcode.de>

diff -ur Xdialog-2.1.1/src/interface.c Xdialog-2.1.1-fixed/src/interface.c
--- Xdialog-2.1.1/src/interface.c	2003-03-25 01:21:58.000000000 +0100
+++ Xdialog-2.1.1-fixed/src/interface.c	2006-02-19 12:07:13.554825000 +0100
@@ -1098,6 +1098,8 @@
 	for (i = 0; i < list_size; i++)
 		glist = g_list_append(glist, options[i]);
 	gtk_combo_set_popdown_strings(GTK_COMBO(combo), glist);
+	if (strlen(Xdialog.default_item) != 0)
+		gtk_entry_set_text(GTK_ENTRY(GTK_COMBO(combo)->entry), Xdialog.default_item);
 
 	gtk_entry_set_editable(GTK_ENTRY(Xdialog.widget1), Xdialog.editable);
 
